File: app/utils/TestXMPPService.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */

package utils;

import com.google.appengine.api.xmpp.JID;
import com.google.appengine.api.xmpp.Message;
import com.google.appengine.api.xmpp.Presence;
import com.google.appengine.api.xmpp.SendResponse;
import java.io.IOException;
import javax.servlet.http.HttpServletRequest;

import com.google.appengine.api.xmpp.XMPPService;

/**
 *
 * @author lstoll
 */
public class TestXMPPService implements XMPPService {
    public static Message lastMessage;
    public static boolean nextAvailable = true;
    public static SendResponse.Status nextSendResult = SendResponse.Status.SUCCESS;
    public static String lastRequestJID;
    public static boolean lastRequestInvitation = false;

    public Presence getPresence(JID jid) {
        reset();
        return new Presence(nextAvailable);
    }

    public Presence getPresence(JID jid, JID jid1) {
        return getPresence(jid);
    }

    public void sendInvitation(JID jid) {
        reset();
        lastRequestInvitation = true;
        lastRequestJID = jid.getId();
    }

    public void sendInvitation(JID jid, JID jid1) {
        sendInvitation(jid);
    }

    public SendResponse sendMessage(Message msg) {
        reset();
        lastMessage = msg;
        lastRequestJID = msg.getRecipientJids()[0].getId(); // TODO = handle multiple case
        SendResponse sr = new SendResponse();
        sr.addStatus(msg.getRecipientJids()[0], nextSendResult);
        return sr;
    }

    public Message parseMessage(HttpServletRequest hsr) throws IOException {
        throw new UnsupportedOperationException("We don't use that here.");
    }

    private void reset() {
        lastMessage = null;
        nextAvailable = true;
        nextSendResult = SendResponse.Status.SUCCESS;
        lastRequestJID = null;
        lastRequestInvitation = false;
    }

}

File: app/utils/TestXMPPService.scala
package utils

import com.google.appengine.api.xmpp.JID
import com.google.appengine.api.xmpp.Message
import com.google.appengine.api.xmpp.Presence
import com.google.appengine.api.xmpp.SendResponse
import java.io.IOException
import javax.servlet.http.HttpServletRequest

import com.google.appengine.api.xmpp.XMPPService

class TestXMPPService extends XMPPService {
    
  
  override def getPresence(jid:JID):Presence = {
    TestXMPPService.reset
    new Presence(TestXMPPService.nextAvailable);
  }
  
  override def getPresence(jid:JID, jid1:JID):Presence = {
    getPresence(jid)
  }
  
  override def sendInvitation(jid:JID) {
    TestXMPPService.reset
    TestXMPPService.lastRequestInvitation = true
    TestXMPPService.lastRequestJID = jid.getId()
  }
  
  override def sendInvitation(jid:JID, jid1:JID) {
    sendInvitation(jid)
  }
  
  override def sendMessage(msg:Message):SendResponse = {
    TestXMPPService.reset
    TestXMPPService.lastMessage = msg
    // TODO = handle multiple case
    TestXMPPService.lastRequestJID = msg.getRecipientJids()(0).getId()
    var sr = new SendResponse
    sr.addStatus(msg.getRecipientJids()(0), TestXMPPService.nextSendResult)
    sr
  }
  
  override def parseMessage(hsr:HttpServletRequest):Message = {
    throw new UnsupportedOperationException("We don't use that here.");
  }
  
  
}

object TestXMPPService {
  var lastMessage:Message = null
  var nextAvailable = true
  var nextSendResult:SendResponse.Status = SendResponse.Status.SUCCESS
  var lastRequestJID:String = null
  var lastRequestInvitation = false
  
  def reset = {
    TestXMPPService.lastMessage = null
    TestXMPPService.nextAvailable = true
    TestXMPPService.nextSendResult = SendResponse.Status.SUCCESS
    TestXMPPService.lastRequestJID = null
    TestXMPPService.lastRequestInvitation = false
  }
}
